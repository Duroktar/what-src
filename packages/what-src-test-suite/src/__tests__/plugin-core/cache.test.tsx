import * as T from '@what-src/plugin-core'

const service = T.getService({
  basedir: '/home/user/app/packages/',
  cache: {},
  options: {},
})

it('caches source locations', () => {
  const testArgs = [
    [{ column: 1, line: 2 }, '/home/user/app/packages/filename1.ts'],
    [{ column: 4, line: 37 }, '/home/user/app/packages/deep/filename2.ts'],
    [{ column: 43, line: 1 }, '/home/user/app/packages/deep/deeper/filename3.ts'],
    [{ column: 3, line: 12 }, '/home/user/app/packages/deep/deeper/deepest/filename4.ts'],
    [{ column: 34, line: 51 }, '/home/user/app/packages/deep/deeper/deepest/filename5.ts'],
  ] as const

  const cacheKeys: string[] = []

  testArgs.forEach(test => {
    cacheKeys.push(service.cache(test[0], test[1]))
  })

  expect<T.WhatSrcCache>(service.getCache()).toMatchSnapshot()
})
