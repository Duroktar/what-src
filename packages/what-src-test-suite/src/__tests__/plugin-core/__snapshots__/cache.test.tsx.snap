// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`caches source locations 1`] = `
Object {
  "1": "{\\"filename\\":\\"filename1.ts\\",\\"remoteFn\\":null,\\"column\\":1,\\"line\\":2}",
  "2": "{\\"filename\\":\\"deep/filename2.ts\\",\\"remoteFn\\":null,\\"column\\":4,\\"line\\":37}",
  "3": "{\\"filename\\":\\"deep/deeper/filename3.ts\\",\\"remoteFn\\":null,\\"column\\":43,\\"line\\":1}",
  "4": "{\\"filename\\":\\"deep/deeper/deepest/filename4.ts\\",\\"remoteFn\\":null,\\"column\\":3,\\"line\\":12}",
  "5": "{\\"filename\\":\\"deep/deeper/deepest/filename5.ts\\",\\"remoteFn\\":null,\\"column\\":34,\\"line\\":51}",
  "__basedir": "/home/user/app/packages/",
}
`;
